src/main/java/frc/robot/superstructure/StatesMotionPlanner.java

```diff
@@ -0,0 +1,27 @@
+package frc.robot.superstructure;
+
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+
+public class StatesMotionPlanner {
+
+	private final Superstructure superstructure;
+
+	public StatesMotionPlanner(Superstructure superstructure) {
+		this.superstructure = superstructure;
+	}
+
+	public Command setState(RobotState state) {
+		return switch (state) {
+			case INTAKE, SPEAKER, AMP, TRANSFER_ELEVATOR_SHOOTER, TRANSFER_SHOOTER_ELEVATOR ->
```

Comment:
All the functions of the states here should start with change state false, and only after the action is done change state true.